File: nodecore-cli/src/main/java/nodecore/cli/serialization/ImportPrivateKeyInfo.java
// VeriBlock NodeCore CLI
// Copyright 2017-2020 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.

package nodecore.cli.serialization;

import com.google.gson.annotations.SerializedName;
import nodecore.api.grpc.VeriBlockMessages;
import nodecore.api.grpc.utilities.ByteStringAddressUtility;

public class ImportPrivateKeyInfo {
    public ImportPrivateKeyInfo(final VeriBlockMessages.ImportPrivateKeyReply reply) {
        this.address = ByteStringAddressUtility.parseProperAddressTypeAutomatically(reply.getResultantAddress());
    }

    @SerializedName("imported_address")
    public String address;
}

File: nodecore-cli/src/main/java/nodecore/cli/serialization/ImportPrivateKeyInfo.kt
// VeriBlock NodeCore CLI
// Copyright 2017-2021 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.
package nodecore.cli.serialization

import com.google.gson.annotations.SerializedName
import nodecore.api.grpc.VeriBlockMessages.ImportPrivateKeyReply
import nodecore.api.grpc.utilities.extensions.toProperAddressType

class ImportPrivateKeyInfo(
    reply: ImportPrivateKeyReply
) {
    @SerializedName("imported_address")
    val address = reply.resultantAddress.toProperAddressType()
}
